### PR TITLE
[nnc] Rename IRSimplifierBase with PolynomialBase

### PR DESCRIPTION
**Pull request by huiguoo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62138
* #60693
* #60688
* #60687
* __->__ #60686

Differential Revision: [D29373316](https://our.internmc.facebook.com/intern/diff/D29373316)

### CODE REVIEW

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29373316).

File: torch/csrc/jit/tensorexpr/ir_simplifier.h [hunk 1]
```diff
@@ -404,9 +404,9 @@ class MinTerm : public ExprNode<MinTerm> {
 };
 
 // Stmt simplification should occur in both modes.
-class TORCH_API IRSimplifierBase : public IRMutator {
+class TORCH_API PolynomialBase : public IRMutator {
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/ir_simplifier.h:407]: What is the need for this change?

Comment (huiguoo) [reply torch/csrc/jit/tensorexpr/ir_simplifier.h:407]: The name indicates that it is the base class of all simplifications, but the context-sensitive simplification (this PR stack) is not based on it. It currently serves as the base class of `PolynomialTransformer` and `TermExpander` so I guess `PolynomialBase` is a more appropriate name?

**Review by navahgar — APPROVED** [review]

**Comment by huiguoo** [comment]
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29373316).

**Pull request closed without merging**